tests/dialogBoxes.spec.ts

```diff
@@ -0,0 +1,27 @@
+import { test, expect } from "@playwright/test";
+
+test("Add and delete pet type", async ({ page }) => {
+    await page.goto("/");
+    await page.getByText(" Pet Types").click();
+    await expect(page.locator("h2")).toHaveText("Pet Types");
+
+    await page.locator('.btn-default', { hasText: " Add " }).click()
+    await expect(page.locator('.xd-container h2').last()).toHaveText('New Pet Type')
+
+    const addNewPetForm = page.locator('.form-group')
+    const inputNewPetType = addNewPetForm.locator('#name')
+    await expect(addNewPetForm.locator('label')).toHaveText("Name")
+    await expect(inputNewPetType).toBeVisible()
+
+    await inputNewPetType.fill('pig')
+    await page.getByRole('button', { name: "Save" }).click()
+
+    page.on('dialog', dialog => {
+        expect(dialog.message()).toEqual('Delete the pet type?')
+        dialog.accept()
+    })
+
+    await page.locator('tr', { has: page.locator('[id="6"]') }).getByRole("button", { name: "Delete" }).click();
```

Comment:
Instead of deleting record with ID 6, it's better to use the similar concept used in the assertion for the next step, where you asserted the last value is not a pig. So the deletion of the element should be also the last one in the list, instead of ID-6.

Reply:
I tried 
 `   await page.locator('tr').getByRole("button", { name: "Delete" }).last().click();`
    
   but is deleting `hamster`(previous) instead of **new** last one (`pig`)

Reply:
is it ok to use `waitForResponce` after click?

Reply:
You can try `await page.getByRole("button", { name: "Delete" }).last().click();`

Reply:
I already added new commit

Reply:
thank you